libs/utils/utils/data.py

```diff
@@ -38,9 +38,12 @@ def calc_shifts_required(Tb: float, T: float, delta: float) -> int:
     combinations in front of the sum above.
     """
 
-    discriminant = (delta / 2 - T) ** 2 - 2 * delta * Tb
-    N = (T - delta / 2 - discriminant**0.5) / delta
-    return math.ceil(N)
+    if Tb > 0:
+        discriminant = (delta / 2 - T) ** 2 - 2 * delta * Tb
+        N = (T - delta / 2 - discriminant**0.5) / delta
+        return math.ceil(N)
+    else:
+        return 0
```

Comment:
The way this is set up, having a negative `Tb` will also return 0, which probably isn't what we want - if someone specified a negative background time, it was probably a typo and so we'd want some kind of error to be raised.

I would just do
```
if Tb == 0:
    return 0
discriminant = ...
```